chiselFrontend/src/main/scala/chisel3/core/Bits.scala

```diff
@@ -550,12 +550,20 @@ private[core] sealed trait UIntFactory {
   // NOTE: This resolves UInt(width = 32)
   def apply(dir: Option[Direction] = None, width: Int): UInt = apply(Width(width))
   /** Create a UInt literal with inferred width.- compatibility with Chisel2. */
-  def apply(value: BigInt): UInt = apply(value, Width())
+  def apply(value: BigInt)(implicit opts: CompileOptions): UInt = {
+    if (opts.deprecateSingleArgumentFactoryMethods) {
+      Builder.deprecated("UInt.Lit(val) or val.U should be used over UInt(val)")
+    }
+    apply(value, Width())
+  }
   /** Create a UInt with a specified direction and width - compatibility with Chisel2. */
-  def apply(dir: Direction, width: Int): UInt = apply(dir, Width(width))
+  def apply(dir: Direction, width: Int)(implicit opts: CompileOptions): UInt = apply(dir, Width(width))(opts)
```

Comment:
Yes. Otherwise the implicit is resolved at the site of the function definition with whatever is currently in scope, which doesn't include the value implicitly passed by the original caller. This may be a Scala bug.